src/heif.h

```diff
@@ -273,6 +273,36 @@ void heif_context_free(struct heif_context*);
 
 struct heif_reading_options;
 
+enum heif_reader_offset {
+  heif_seek_start,
+  heif_seek_current,
+  heif_seek_end
+};
+
+struct heif_reader {
+  // API version supported by this reader
+  int reader_api_version;
+
+  // --- version 1 functions ---
+  uint64_t (*get_length)(struct heif_context* ctx, void* userdata);
+  uint64_t (*get_position)(struct heif_context* ctx, void* userdata);
```

Comment:
As seek() takes an int64, we probably also should return an int64 here.

Reply:
As the position of a file can never be negative, I would rather keep the `uint64_t`. `seek()` allows negative numbers to seek backwards.

Reply:
I'm afraid of all the compiler warnings that will show up: "passing unsigned value to signed parameter may lose data". 63 bit for file length should be enough.

Reply:
Ok, changed.